### PR TITLE
[CSL-2208] Migrate Quizzes /finalize endpoint to new /results endpoint

### PR DESCRIPTION
**Pull request by esezen**

### CODE REVIEW

File: src/types/quizzes.d.ts [hunk 1]
```diff
@@ -9,6 +20,13 @@ export interface QuizzesParameters {
   versionId?: string;
 }
 
+export interface QuizzesResultsParameters extends QuizzesParameters {
+  answers: any[];
```
Comment (mocca102) [inline src/types/quizzes.d.ts:24]: answers already part of QuizzesParameters did you intend to have it here also

File: src/modules/quizzes.js [hunk 2]
```diff
@@ -140,10 +154,13 @@ class Quizzes {
...
    * @param {string} id - The identifier of the quiz
-   * @param {string} [parameters] - Additional parameters to refine result set
+   * @param {string} parameters - Additional parameters to refine result set
+   * @param {array} parameters.answers - An array of answers in the format [[1,2],[1]]
    * @param {string} [parameters.section] - Product catalog section
-   * @param {array} [parameters.answers] - An array of answers in the format [[1,2],[1]]
    * @param {string} [parameters.versionId] - Specific version identifier for the quiz
+   * @param {number} [parameters.page] - The page number of the results
+   * @param {number} [parameters.resultsPerPage] - The number of results per page to return
+   * @param {object} [parameters.filters] - Key / value mapping (dictionary) of filters used to refine results
```
Comment (mocca102) [inline src/modules/quizzes.js:163]: we're missing the quiz_session_id as a parameter

File: src/modules/quizzes.js [hunk 3]
```diff
@@ -35,12 +35,12 @@ function createQuizUrl(quizId, parameters, options, path) {
...
   }
 
-  if (path === 'finalize' && (typeof parameters.answers !== 'object' || !Array.isArray(parameters.answers) || parameters.answers.length === 0)) {
+  if (path === 'results' && (typeof parameters.answers !== 'object' || !Array.isArray(parameters.answers) || parameters.answers.length === 0)) {
     throw new Error('answers is a required parameter of type array');
   }
 
   if (parameters) {
-    const { section, answers, versionId } = parameters;
+    const { section, answers, versionId, page, resultsPerPage, filters } = parameters;
```
Comment (mocca102) [inline src/modules/quizzes.js:43]: reminder to destruct quiz_session_id here as well?

File: spec/src/modules/quizzes.js [hunk 4]
```diff
@@ -229,10 +229,13 @@ describe(`ConstructorIO - Quizzes${bundledDescriptionSuffix}`, () => {
...
+        expect(res).to.have.property('request').to.be.an('object');
+        expect(res).to.have.property('response').to.be.an('object');
+        expect(res).to.have.property('result_id').to.be.an('string');
         expect(res).to.have.property('version_id').to.be.an('string');
+        expect(res).to.have.property('quiz_id').to.be.an('string');
         expect(fetchSpy).to.have.been.called;
+        expect(res.response).to.have.property('results').to.be.an('array');
         expect(requestedUrlParams).to.have.property('key');
         expect(requestedUrlParams).to.have.property('i');
         expect(requestedUrlParams).to.have.property('s');
```
[the remark this answers is not in the document]
Comment (mocca102) [reply spec/src/modules/quizzes.js:241]: Also as I remember that if we provide the quiz_session_id as a parameter it should be returned in the response. If what I'm remembering is correct can we add a test case for this as well?

**Review by mocca102 — COMMENTED** [review]
Main functionality is working great. Thank you for adding this and giving great attention to types. Just added some comments. let me know what you think

File: spec/src/modules/quizzes.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (esezen) [reply spec/src/modules/quizzes.js:241]: Should be handled here https://github.com/Constructor-io/constructorio-client-javascript/pull/245

File: src/types/quizzes.d.ts (hunk 1 above)
Comment (esezen) [reply src/types/quizzes.d.ts:24]: They are optional there but required here. Is this the right way to do it?

File: src/modules/quizzes.js (hunk 2 above)
Comment (esezen) [reply src/modules/quizzes.js:163]: Handled here https://github.com/Constructor-io/constructorio-client-javascript/pull/245/

File: src/modules/quizzes.js (hunk 3 above)
Comment (esezen) [reply src/modules/quizzes.js:43]: Handled here https://github.com/Constructor-io/constructorio-client-javascript/pull/245/

File: src/types/quizzes.d.ts (hunk 1 above)
Comment (mocca102) [reply src/types/quizzes.d.ts:24]: Looks right to me didn't notice that it's required here

**Review by jjl014 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**